src/plugins/data/public/query/query_string/dataset_service/lib/index_pattern_type.ts

```diff
@@ -12,24 +12,27 @@ import {
   Dataset,
   IIndexPattern,
   DATA_STRUCTURE_META_TYPES,
+  DataStructureCustomMeta,
 } from '../../../../../common';
 import { DatasetTypeConfig } from '../types';
 import { getIndexPatterns } from '../../../../services';
 
-export const indexPatternTypeConfig: DatasetTypeConfig = {
-  id: DEFAULT_DATA.SET_TYPES.INDEX_PATTERN,
-  title: 'Index Patterns',
-  meta: {
+class IndexPatternTypeConfig extends DatasetTypeConfig {
```

Comment:
why do we want to change it to a class based approach over the simpler config? I like the config because i dont have to reimplement a class per language. Classes are useful when you want to make multiple objects of the same class or if you want to reference a singleton. This also differs from the pattern we have in OSD to register stuff. Even registering a whole app does not require a class.